test/ranking.spec.js

```diff
@@ -0,0 +1,112 @@
+import {
+    Ranking
+} from '../src/app/Ranking';
+
+describe("Ranking's logic", () => {
+    const peopleRanking = new Ranking('people');
+
+    const results = [{
+        'user': 'user1',
+        'score': 15,
+        'maxScore': 30 // 50%
+    }, {
+        'user': 'user2',
+        'score': 18,
+        'maxScore': 40 // 45%
+    }, {
+        'user': 'user3',
+        'score': 15,
+        'maxScore': 25 // 60%
+    }];
+
+    afterEach(() => {
+        localStorage.clear();
+    });
+
+    it('When set wrong category name, exception will be thrown', () => {
+        expect(() => {
+            new Ranking('vehicle');
+        }).toThrow('"vehicle" mode is not valid!');
+    });
+
+    it('When new score is higher than or equal to at least one of top results, return true', () => {
+        localStorage.setItem('people', JSON.stringify(results));
+        const score = 9;
+        const maxScore = 18; // 45%
+
+        expect(peopleRanking._isHigherThanCurrentTop(score, maxScore)).toBeTruthy;
```

Comment:
Tutaj niestety to bardzo ważna kwestia i muszę się przyczepić :) 
JavaScript jest trochę ułomny i nie ma metod "prywatnych". Rozumiem, że oznaczyłaś je jako "prywatne" poprzez zastowoanie `_`. W takim razie skoro one są prywatne to nie powinno się ich używać poza daną klasą. Tak samo w teście nie testujemy ich. To ma swoje uzasadnienie - metoda prywatna jest sposobem organizacji kodu - dla czytelności. Możemy dodawać je / pozbywać się co nie powinno nam zmieniać zachowania metod publicznych i wpływać na inną część programu. Skoro zmieniamy sposób organizacji kodu, a nie zachowanie klasy nie powinny też failować żadne testy. Dlatego powinniśmy pozbyć się testów metod prywatnych a testować działanie komponentu jedynie poprzez te dostępne dla wszystkich - publiczne. Test powinien pokazywać też właśnie jak się używa danej klasy w różnych przypadkach - być żywą dokumentacją. Czyli tutaj tak naprawdę test powinien zapisywać jakieś wyniki i sprawdzać kilka przypadków (więcej niż 3, mniej niż 3 itp. ) jakie wyniki zostaną zwrócone.

Reply:
Czyli mogę usunąć wszystkie testy dla metod "prywatnych"? Dodałam je właściwie dla swojej wygody, z ich implementacją - operacjami na listach - miałam najwięcej problemów, a stanowią podstawę metod publicznych, więc od nich zaczęłam i tak już zostały :)

Reply:
Tak, dobrze to usunąć :) W procesie implementacji czasem pisze się mniejsze testy - więc takie "dla wygody" to dobra praktyka - szczególnie przy cięższych przypadkach. Tylko zastanawiam się jak one Ci pomogły skoro się nie odpalały asercje :D Potem jest refactor i niektóre testy wylatują, jeśli były dla prywatnych metod. Testując pokazujemy też jak używać naszego kodu - mamy żywą dokumentację. A nikt nie powinien już tych metod uzywać :) Dobrze wtedy jakby te testy na metody publiczne wtedy pokazywały różne przypadki. Takie, żeby do wszystkich ifów powchodziło.

Reply:
Done